src/xscen/catalog.py

```diff
@@ -187,7 +187,10 @@ def __init__(
             if datecol in self.df.columns and self.df[datecol].dtype == "O":
                 # Missing values in object columns are np.nan, which numpy can't convert to datetime64 (what's up with that numpy???)
                 self.df[datecol] = (
-                    self.df[datecol].fillna("").to_numpy().astype("datetime64[ms]")
+                    self.df[datecol]
+                    .infer_objects(copy=False)
+                    .dropna()
+                    .astype("datetime64[ms]")
```

Comment:
```suggestion
                    self.df[datecol]
                    .infer_objects(copy=False)
                    .to_numpy()
                    .astype("datetime64[ms]")
```
I'd like a second opinion from @aulemahal, but I think that this is the right fix. We don't want to drop empty entries, as they might come from `fx` frequencies, but convert them to NaT. This seems to do the right thing.

Minimal example:
`pd.Series(np.array([5,5,None,5])).astype("object").infer_objects(copy=False).to_numpy().astype("datetime64[ms]")`
`array(['1970-01-01T00:00:00.005',  '1970-01-01T00:00:00.005',  'NaT', '1970-01-01T00:00:00.005'], dtype='datetime64[ms]')`

Reply:
I'm not sure I understand what `infer_objects` is supposed to do, but @Zeitsperre's solution works as the index is preserved on the series even after dropping the NaNs. And pandas understand that it must put `NaT` when merging it back.

```suggestion
                    self.df[datecol]
                    .dropna()
                    .astype("datetime64[ms]")
```

Seems that the `to_numpy` is not required anymore but don't know when it changed (I am using pandas 2.2.3).  Also, I didn't get any warnings in my tests with the previous method, do you remember what warnings we are addressing here ?

Reply:
There is no warning when no data is missing, but I get the warning with:

`pd.Series(np.array([5,5,5])).astype("object").fillna("").to_numpy().astype("datetime64[ms]")`

`FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
`

Reply:
I didn't get the warning with missing data. In a catalog's csv, the `date_start` and `date_end` column are expected to be strings or "nan", which in theory would never trigger a "downcasting". But anyway, that step seems unnecessary now.